pkg/telemetry/metrics_db_lock.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package telemetry
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/common"
+
+	"github.com/juju/fslock"
+)
+
+const (
+	LocalTanzuCLIMetricsDBFileLock = ".cli_metrics_db.lock"
+	// DefaultMetricsDBLockTimeout is the default time waiting on the filelock
+	DefaultMetricsDBLockTimeout = 10 * time.Minute
```

Comment:
IMO, 10 mins is too high!! 
How about configuring it with env variable or config!

Reply:
Good point.I would reduce it to 2 seconds to make sure it would not add latency to the user commands as telemetry is best-effort and should not affect the user experience on CLI/plugin functionality